cw/cw305/tvla.py

```diff
@@ -0,0 +1,1016 @@
+        from the data in the ChipWhisperer project file. Ignored for number-of-steps > 1.""",
+    )
+    parser.add_argument(
+        "-d",
+        "--save-to-disk",
+        action="store_true",
+        default=False,
+        help="""Save trace, leakage and t-test files to disk. Ignored for trace and leakage files
+        when number-of-steps > 1.""",
+    )
+    parser.add_argument(
+        "-r",
+        "--round-select",
+        help="""Index of the AES round for which the histograms are to be computed: 0-10. Not
+        required. If not provided, the histograms for all AES rounds are computed.""",
+    )
+    parser.add_argument(
+        "-b",
+        "--byte-select",
+        help="""Index of the AES state byte for which the histograms are to be computed: 0-15. Not
+        required. If not provided, the histograms for all AES state bytes are computed.""",
+    )
+    parser.add_argument(
+        "-i",
+        "--input-file",
+        help="""Name of the input file containing the histograms. Not required. If both -i and -o
+        are provided, the input file is appended with more data to produce the output file.""",
+    )
+    parser.add_argument(
+        "-o",
+        "--output-file",
+        help="""Name of the output file to store generated histograms. Not required. If both -i and
+        -o are provided, the input file is appended with more data to produce the output file.""",
+    )
+    parser.add_argument(
+        "-n",
+        "--number-of-steps",
+        type=int,
+        default="1",
+        help="""Number of steps to breakdown the analysis into. For every step, traces are
+        separately filtered and the leakage is computed. The histograms are appended to the
+        ones of the previous step. This is useful when operating on very large trace sets and/or
+        when analyzing how results change with the number of traces used.""",
+    )
+    parser.add_argument(
+        "-a",
+        "--ttest-step-file",
+        help="""Name of the t-test step file containing one t-test analysis per step. Not
+        required. If not provided, the data is recomputed.""",
+    )
+    parser.add_argument(
+        "-f",
+        "--plot-figures",
+        action="store_true",
+        default=False,
+        help="""Plot figures and save them to disk. Not required.""",
+    )
+    parser.add_argument(
+        "-g",
+        "--general-test",
+        action="store_true",
+        default=False,
+        help="""Perform general fixed-vs-random TVLA without leakage model. Odd traces are grouped
+        in the fixed set while even traces are grouped in the random set. Not required.""",
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        default="aes",
+        help="""Select mode: can be either "aes" or "sha3". Not required. If not provided or if a
+        another string is provided, "aes" is used.""",
+    )
+
+    return parser.parse_args()
+
+
+def main():
```

Comment:
nit: this function could benefit from some refactoring (e.g., moving chunks of code to helper functions).